flashinfer/api_logging.py

```
@@ -82,11 +108,947 @@ def _setup_logger():
+                    from safetensors.torch import save_file
+
+                    # safetensors requires contiguous tensors
+                    tensors_contiguous = {
+                        k: v.contiguous() for k, v in input_tensors.items()
+                    }
+                    save_file(tensors_contiguous, str(dump_dir / "inputs.safetensors"))
+                except ImportError:
+                    _logger.error(
+                        "safetensors package not installed. "
+                        "Install with: pip install safetensors"
+                    )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)